src/cocotb/_base_triggers.py

```diff
@@ -289,20 +329,18 @@ def __init__(self, parent: "Lock") -> None:
         super().__init__()
         self._parent = parent
 
-    def _prime(self, callback: Callable[["Self"], None]) -> None:
-        if self._primed:
-            raise RuntimeError(
-                "Lock.acquire() result can only be used by one task at a time"
-            )
-        self._callback = callback
+    def _prime(self) -> None:
         self._parent._prime_lock(self)
-        return super()._prime(callback)
 
     def _unprime(self) -> None:
-        if not self._primed:
-            return
         self._parent._unprime_lock(self)
-        return super()._unprime()
+
+    def __await__(self) -> "Generator[Self, None, Self]":
+        if self._parent._is_used(self):
+            raise RuntimeError(
+                "Lock.acquire() result can only be used by one task at a time"
```

Comment:
Would it be helpful to print which task already has the lock?

Reply:
Yeah. But we don't have this info here. Triggers are currently agnostic to Tasks.